clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -349,7 +351,16 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
 
         if (numHeaders == 0)
             return new DefaultRecord(sizeInBytes, attributes, offset, timestamp, sequence, key, value, Record.EMPTY_HEADERS);
+        try {
```

Comment:
Hard one. I have already changed this to cover the full method, but @ijuma requested it to be widened again to only this part. You are probably right, that more generic exception, conveying also the failures in the wider function will allow the try to span the whole function. Let me know.